pkg/middleend/middleend.go

```diff
@@ -304,3 +294,32 @@ func (s *Server) EncryptedVolumeStats(_ context.Context, in *pb.EncryptedVolumeS
 		UnmapLatencyTicks: int32(result.Bdevs[0].UnmapLatencyTicks),
 	}}, nil
 }
+
+func (s *Server) getAccelCryptoKeyCreateParams(volume *pb.EncryptedVolume) models.AccelCryptoKeyCreateParams {
+	var params models.AccelCryptoKeyCreateParams
+
+	switch volume.Cipher {
+	case pb.EncryptionType_ENCRYPTION_TYPE_AES_CBC_128:
+		fallthrough
+	case pb.EncryptionType_ENCRYPTION_TYPE_AES_CBC_192:
+		fallthrough
+	case pb.EncryptionType_ENCRYPTION_TYPE_AES_CBC_256:
+		params.Cipher = "AES_CBC"
+		params.Key = hex.EncodeToString(volume.Key)
+		params.Key2 = ""
+
+	case pb.EncryptionType_ENCRYPTION_TYPE_AES_XTS_128:
+		fallthrough
+	case pb.EncryptionType_ENCRYPTION_TYPE_AES_XTS_192:
+		fallthrough
+	case pb.EncryptionType_ENCRYPTION_TYPE_AES_XTS_256:
+		params.Cipher = "AES_XTS"
+		keyHalf := len(volume.Key) / 2
```

Comment:
I think we also need to check key length here. Spdk will return error on invalid key size

Reply:
Ok, I will provide an additional validation